frontend/packages/ceph-storage-plugin/src/components/dashboards/persistent-internal/details-card.tsx

```diff
@@ -110,14 +146,36 @@ const DetailsCard: React.FC<DashboardItemProps> = ({
             {infrastructurePlatform}
           </DetailItem>
           <DetailItem title={t('ceph-storage-plugin~Mode')}>Internal</DetailItem>
-          <DetailItem
-            key="version"
-            title={t('ceph-storage-plugin~Version')}
-            isLoading={!subscriptionLoaded}
-            error={subscriptionLoaded && !ocsVersion}
-          >
-            {ocsVersion}
-          </DetailItem>
+          {!isEmpty(filteredVersions) &&
+          ocsSubscription?.[0]?.spec?.installPlanApproval === 'Manual' ? (
+            <DetailItem
+              key="version"
+              title={t('ceph-storage-plugin~Version')}
+              isLoading={!subscriptionLoaded}
+              error={subscriptionLoaded && !ocsVersion}
+            >
+              {ocsVersion}
+              <Button
+                type="button"
+                isInline
+                onClick={ocsChannelModal}
+                variant="link"
+                isDisabled={!ocsSubscriptionLoaded || !packageManifest}
+              >
+                <BlueArrowCircleUpIcon className="co-icon-space-r" />
+                {ocsSubscriptionLoaded ? ocsSubscription?.[0].spec.channel : null}
+              </Button>
```

Comment:
```suggestion
             !isAutomatic && <Button
                type="button"
                isInline
                onClick={ocsChannelModal}
                variant="link"
                isDisabled={!ocsSubscriptionLoaded || !packageManifest}
              >
                <BlueArrowCircleUpIcon className="co-icon-space-r" />
                {ocsSubscriptionLoaded ? ocsSubscription?.[0].spec.channel : null}
              </Button>
```
Rather than disabling this button hide it until it's ready to be shown. So until the subscription and pkgManifest is gets loaded hide the link.